kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -99,6 +137,21 @@ private[spark] class KubernetesClusterSchedulerBackend(
     // create a single k8s executor pod.
     val labelMap = Map("type" -> "spark-executor")
     val podName = s"$sparkExecutorName-$executorNum"
+
+    val submitArgs = mutable.ArrayBuffer.empty[String]
+
+    if (conf.getBoolean("spark.dynamicAllocation.enabled", false)) {
+      submitArgs ++= Vector(
+        "dynamic-executors")
```

Comment:
My original plan was to just detect the `--conf spark.dynamicAllocation.enabled` flag, but the `ExecutorBackend` doesn't recognize `--conf` args (it appears to expect its conf sent from the driver), so I needed something easy to strip off the argument list.  Alternatively, the pod could be configured with some additional environment variable, but that isn't any simpler, afaict.